packages/kyc-oracle/KycOracle.js

```diff
@@ -0,0 +1,96 @@
+const axios = require("axios");
+const ethers = require("ethers");
+const express = require("express");
+
+class KycOracle {
+  /**
+   * Constructor for ReputationMiner
+   * @param {string} privateKey              The private key of the address that has the administration permission.
+   *                                           If used, `userAddress` is not needed and will be derived.
+   * @param {string} userAddress             The address that has the administration permission to edit the whitelist.
+   * @param {string} whitelistAddress        The address the whitelist.
+   * @param {Object} loader                  The loader for loading the contract interfaces. Usually a TruffleLoader.
+   * @param {Object} provider                Ethers provider that allows access to an ethereum node.
+   * @param {Number} [port]                  The port the oracle will serve on
+   */
+  constructor({ privateKey, userAddress, apiKey, loader, provider, port = 3000 }) {
+    if (privateKey) {
+      this.wallet = new ethers.Wallet(privateKey, this.provider);
+      this.userAddress = this.wallet.address;
+    } else {
+      this.wallet = provider.getSigner(userAddress);
+      this.userAddress = userAddress;
+    }
+    console.log("Transactions will be signed from ", this.userAddress);
+
+    this.apiKey = apiKey;
+
+    this.loader = loader;
+    this.provider = provider;
+
+    this.app = express();
+
+    this.app.use(function (req, res, next) {
+      res.header("Access-Control-Allow-Origin", "*");
```

Comment:
Seems like if this is going to be an API exposed to the world, we should do better than this.

Reply:
I suppose restricting access to the dapp would be appropriate.